src/types/UiaTracing.cpp

```diff
@@ -0,0 +1,523 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+#include "UiaTracing.h"
+
+// we need to disable a few warnings because
+// the TraceLogging code is out of our control
+#pragma warning(push)
+#pragma warning(disable : 26446 26447 26477 26482 26485 26494 26496)
+TRACELOGGING_DEFINE_PROVIDER(g_UiaProviderTraceProvider,
+                             "Microsoft.Windows.Console.UIA",
+                             // tl:{e7ebce59-2161-572d-b263-2f16a6afb9e5}
+                             (0xe7ebce59, 0x2161, 0x572d, 0xb2, 0x63, 0x2f, 0x16, 0xa6, 0xaf, 0xb9, 0xe5));
+
+using namespace Microsoft::Console::Types;
+
+UiaTracing::UiaTracing() noexcept
+{
+    TraceLoggingRegister(g_UiaProviderTraceProvider);
+}
+
+UiaTracing::~UiaTracing() noexcept
+{
+    TraceLoggingUnregister(g_UiaProviderTraceProvider);
+}
+
+inline std::wstring UiaTracing::_getValue(const ScreenInfoUiaProviderBase& /*siup*/) noexcept
+try
+{
+    std::wstringstream stream;
+    stream << " NO IDENTIFYING DATA";
+    return stream.str();

```

Comment:
Uh... can we just return the string until proven otherwise instead of making a whole string stream and coming back to the string?